akka-remote/src/main/scala/akka/remote/RemoteMetricsExtension.scala

```diff
@@ -51,18 +55,38 @@ private[akka] class RemoteMetricsOff extends RemoteMetrics {
   override def logPayloadBytes(msg: Any, payloadBytes: Int): Unit = ()
 }
 
+/**
+ * INTERNAL API
+ */
+private object RemoteMetricsOn {
+  val LogFrameSizeExceedingConfigKey = "akka.remote.log-frame-size-exceeding"
+}
+
```

Comment:
I think a better way to do this is to read the config in `RemoteSettings`. Make it an `Option[Int]`. Then it will fail fast when the ActorSystem is started. We usually think it's better to fail the startup completely rather than falling back to a silent behavior.

Then in `RemoteMetricsExtension.createExtension` the `RemoteSettings` can be accessed via `RARP(system).provider.remoteSettings`.